src/core/Akka.Remote/PhiAccrualFailureDetector.cs

```diff
@@ -371,7 +354,7 @@ public double StdDeviation
         {
             if (history._intervals.Count < history._maxSampleSize)
             {
-                return new HeartbeatHistory(history._maxSampleSize, history._intervals.Concat(new[] { interval }).ToList(),
+                return new HeartbeatHistory(history._maxSampleSize, history._intervals.Add(interval),
```

Comment:
Use Systems.Collections.Immutable here instead of classic collections and LINQ (this is 7-8 year old code)